pkg/controller/service/servicecontroller.go

```diff
@@ -125,16 +127,33 @@ func New(
 		nodeLister:       nodeInformer.Lister(),
 		nodeListerSynced: nodeInformer.Informer().HasSynced,
 		workingQueue:     workqueue.NewNamedDelayingQueue("service"),
+		endpoints: make(map[string]int),
 	}
 
 	serviceInformer.Informer().AddEventHandlerWithResyncPeriod(
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: s.enqueueService,
 			UpdateFunc: func(old, cur interface{}) {
+				endpointsChanged := false
 				oldSvc, ok1 := old.(*v1.Service)
 				curSvc, ok2 := cur.(*v1.Service)
```

Comment:
Can these checks be includes as part of 'needsUpdate' function? That way we consolidate the decision making through that interface.